contracts/orderbook/Logic/OrderBookOrderPlace.sol

```diff
@@ -0,0 +1,258 @@
+pragma solidity ^0.8.0;
+import "../Storage/OrderBookEvents.sol";
+import "../Storage/OrderBookStorage.sol";
+import "../OrderVault/IDeposits.sol";
+
+contract OrderBookOrderPlace is OrderBookEvents, OrderBookStorage {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    function initialize(address target) public onlyOwner {
+        _setTarget(this.placeOrder.selector, target);
+        _setTarget(this.amendOrder.selector, target);
+        _setTarget(this.cancelOrder.selector, target);
+        _setTarget(this.cancelOrderProtocol.selector, target);
+        _setTarget(this.changeStopType.selector, target);
+        _setTarget(this.recoverFundsFromFailedOrder.selector, target);
+    }
+
+    function queryRateReturn(
+        address start,
+        address end,
+        uint256 amount
+    ) public view returns (uint256) {
+        (uint256 executionPrice, uint256 precision) = IPriceFeeds(
+            protocol.priceFeeds()
+        ).queryRate(start, end);
+        return (executionPrice * amount) / precision;
+    }
+
+    function _caseChecks(bytes32 ID, address collateral, address loanToken)
+        internal
+        view
+        returns (bool)
+    {
+        return 
+            protocol.getLoan(ID).loanToken ==
+            loanToken &&
+            protocol.getLoan(ID).collateralToken ==
+            collateral &&
+            protocol.delegatedManagers(ID, address(this));
+    }
+
+    function _isActiveLoan(bytes32 ID) internal view returns (bool) {
+        return protocol.loans(ID).active;
+    }
+
+    function _abs(int256 x) private pure returns (int256) {
+        return x >= 0 ? x : -x;
+    }
+
+    function placeOrder(IOrderBook.Order calldata Order) external pausable {
+        require(Order.loanDataBytes.length < 3500, "OrderBook: loanDataBytes too complex");
+        require(
+            _abs(
```

Comment:
I think there is an easier way to check
you have to check here either loanTokenAmount > 0 or collateralTokenAmount > 0 but can't be both >0
`require(!(collateralTokenAmount>0 && loanTokenAmount >0))`
or
`require(!collateralTokenAmount>0 || !loanTokenAmount >0)`
I think this is cheaper

Reply:
yes. will implement